runtime/sema/type.go

```diff
@@ -3761,6 +3792,10 @@ func (t *CompositeType) GetNestedTypes() *StringTypeOrderedMap {
 	return t.nestedTypes
 }
 
+func (t *CompositeType) SetNestedTypes(nestedTypes *StringTypeOrderedMap) {
+	t.nestedTypes = nestedTypes
+}
```

Comment:
I needed a way to set the nested types when decoding. There are a couple of other options though:
1. From what I can tell, the only CompositeTypes with nested types are built-ins. These could be encoded as enums instead of encoded themselves.
2. To account for recursive types, it is necessary to save the CompositeType in the decoder's typeDefs map before decoding further. A deferral mechanism can be added so the CompositeType can be referred to during decoding before it is instantiated.